### DISCUSSION TITLE
Pushbox, possibility to set different values for several states.

### DISCUSSION THREAD

**Discussion by GameHackFan** [ikemen-engine/Ikemen-GO#881] [Feature Request] [answered]
Hi, I noticed that on mugen you have `ground.back` and `ground.front` to change the pushbox width and `height` to change its height. I am requesting the possibility to change its width and height based on the character state, like crouch or jump or any other state. I know that you can change the width of the pushbox for some scenarios, but the height is just that same value, you need some code to handle that, it would be nice to have a way to set its value based on the animation or the state, so we can easily have different width and height for crouch, liedown, air, etc.

**Comment by Sluigi123** [comment]
So, here are the ones we have currently, in regards to pushboxes:
_ground.back
ground.front
air.back
air.front
height_

And here's what we should have along with them, assuming to be within GameHackFun's viewing correctly:
'ground.back' -> 'ground.back.stand', 'ground.back.crouch', 'ground.back.down'
_-- While making 'ground.back' being applied as the default to all three of them._
'ground.front' -> 'ground.front.stand', 'ground.front.crouch', 'ground.front.down'
_-- While making 'ground.front' being applied as the default to all three of them._
Let 'air.back' & 'air.front' be left as is; these are actually fine.
'height' -> 'height.stand', 'height.crouch', 'height.down'
_-- While making 'height' being applied as the default to all three of them._
Add one, called 'height.air.lw.pos' and 'height.air.hi.pos', where it moves the pushbox up or down for positive or negative values respectively. Defaults to 'height' (or 'height.stand' when applied) and 0 respectively.
Like in almost all fighting games, characters always have smaller pushboxes and are positioned higher then normal, while in midair; this is more-so on large characters, such as Zangief or Potemkin.

Let me know if I'm missing anything else.

**Comment by GameHackFan** [comment]
Yeah, I think crouch, stand and air are the main ones, some games also have different boxes when the enemy is knocked down, but the main 3 that I feel is a must is stand, crouch and air

**Comment by potsmugen** [comment, marked as answer]
Implemented in `stand.sizebox`, `crouch.sizebox`, `air.sizebox` and `down.sizebox` constants.
